lib/vdsm/virt/backup.py

```diff
@@ -355,13 +355,14 @@ def _start_monitoring_scratch_disks(vm, backup_disks, backup):
     disks = backup["disks"]
     for backup_disk in backup_disks.values():
         if backup_disk.scratch_disk.type == DISK_TYPE.BLOCK:
-            # TODO: Add domain, image, and volume ids when they are
-            # availble.
             scratch_disk = disks[backup_disk.drive.name]
             vm.log.info("Start monitoring scratch disk %s for drive %s",
                         scratch_disk, backup_disk.drive.name)
             backup_disk.drive.scratch_disk = {
                 "index": scratch_disk["index"],
+                "sd_id": backup_disk.scratch_disk.sd_id,
+                "img_id": backup_disk.scratch_disk.img_id,
+                "vol_id": backup_disk.scratch_disk.vol_id,
```

Comment:
Change looks good now, but it will add:

    "sd_id": None

When engine did not send the uuids. In this case we cannot monitor the scratch disk
so we don't need the ids in the dict.

Lets add the ids only of the they are not None. For example:

    if backup_disk.scratch_disk.sd_id is not None:
        backup_disk.drive.scratch_disk["sd_id"] = backup_disk.scratch_disk.sd_id

This also probably breaks backup tests, checking for the old
content of the dict.

I guess you need to update the tests - we should test:
- old way: engine sends no scratch disk ids
- new way: engine sends scratch disks ids

You can run the backup tests locally with:

    tox -e virt tests/virt/backup_test.py

The tests run very quickly, no need to push changes to github and wait 5 minutes until
the all the tests pass.

Reply:
> Fixed according to your suggestion. While looking for a more elegant solution encountered the following arguably better implementation :)
> 
> ```
> {k: v for k, v in (('orange', orange), ('apple', apple)) if v is not None}
> ```

This works well for converting list of tuples to dict, removing None values. But we
don't have such list, and the simple approach is simpler.